re_data_ui/src/components/LineageGraph.tsx

```diff
@@ -45,9 +59,19 @@ const LineageGraph: React.FC<LineageGraphProps> = (
   );
 };
 
-export default memo(LineageGraph, (prevState) => {
-  if (!prevState.overviewDataLoaded) return false; // allow re-renders if we've not loaded the graph
-  // Return true indicating that the props are always
-  // equal to avoid re-rendering of lineage graph since it's static
-  return true;
+// eslint-disable-next-line consistent-return
+export default memo(LineageGraph, (prevState, nextState) => {
+  const { overviewDataLoaded, showModelDetails, data: prevData } = prevState;
+  const { data } = nextState;
+  if (showModelDetails) {
+    if (!overviewDataLoaded) return false; // allow re-renders if we've not loaded the graph
+    // allow re-renders if we've not loaded the graph
+    // Return true indicating that the props are always
+    // equal to avoid re-rendering of lineage graph since it's static
+    return true;
+  } if (!showModelDetails) {
+    if (JSON.stringify(prevData) !== JSON.stringify(data)) return false;
+    return true;
+  }
+  return false;
```

Comment:
do we still need to control the re rendering using this `propsAreEqual` function? Now that we have a function that clears nodes and edges before adding inside the `LineageGraph` component.

Reply:
i used that as a way to double check but on second thought, there's actually no need

Reply:
@dejii removing this causes re-render once a node has been selected

Reply:
that's strange.

Reply:
i re-checked

removing this prevent the data props from being updated. @dejii

Reply:
👍